src/targets/gpu/include/migraphx/gpu/where.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_WHERE_HPP
+#define MIGRAPHX_GUARD_RTGLIB_WHERE_HPP
+
+#include <migraphx/gpu/oper.hpp>
+#include <migraphx/gpu/device/where.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace gpu {
+
+struct hip_where : ternary_device<hip_where, device::where>
+{
+    shape compute_shape(const std::vector<shape>& inputs) const
+    {
+        check_shapes{inputs, *this}.has(4).same_dims();
```

Comment:
Shape of the first input also needs to be considered, so the following logic is not enough.

Reply:
To avoid the complex scenarios, we can compute the output shape as (incorrect syntax):

```
        if(s1 == s2 == s3 and s1.packed())
        {
            return s1;
        }
        else
        {
            return {s1.type(), s1.lens()};
        }
```
And then in the parse_where, we need to broadcast the input, so their dims are the same.

Reply:
> Shape of the first input also needs to be considered

It is, `.same_dims` will ensure that the condition has the same dimensions as the other two args. 

> To avoid the complex scenarios, we can compute the output shape as

We dont want to do that as we will lose the permutation of input.

Reply:
But just following the binary ops scenarios are not enough since it misses the cond, which can also be transposed or broadcasted.

Reply:
But why does it matter if the cond is transposed or broadcasted? Its not the tensor that will be returned. So if the cond is nchw and one input is nhwc and the other is braodcasted, then we want the output to be nhwc(not nchw as that shouldn't be considered at all).